summarization/stamp_page_picking/interesting_sequence_picker.py

```diff
@@ -0,0 +1,233 @@
+                    and self.stamp_page_covers[index].cover[i] == 1:
+                total_weight += 1
+        return total_weight / self.stamp_page_covers[index].cost
+
+    def _compute_weighted_score(self, score_list):
+        return sum(
+            [a * b for a, b in zip(score_list, self.individual_score_weights)]
+        )
+
+    def _get_approximate_index_for_stamp_page(self, stamp_page_index):
+        '''Fetches the approximate index so it can be used
+        for computing its distance to the sweepeing index
+        '''
+        index_sum = 0
+        valid_index_count = 0
+        stamp_page = self.stamp_pages[stamp_page_index]
+
+        if stamp_page.media_index != -1:
+            index_sum += stamp_page.media_index
+            valid_index_count += 1
+
+        if stamp_page.sentence_index != -1:
+            index_sum += stamp_page.sentence_index
+            valid_index_count += 1
+
+        return index_sum / valid_index_count
+
+    def _get_distance_from_sweeping_index(self, stamp_page_index):
+        approximate_index = self._get_approximate_index_for_stamp_page(
+            stamp_page_index)
+        return abs(self.sweeping_index - approximate_index)
+
+    def _interestingess_metric(self, stamp_page_index):
+        '''
+        calculates the interesting-ness metric based
+        on the last picked stamp page
+        '''
+        from_stamp_page_index = self.last_picked_stamp_page_index
+        to_stamp_page_index = stamp_page_index
+
+        content_change_score \
+            = self._get_content_change_score(
+                from_stamp_page_index,
+                to_stamp_page_index
+            )
+
+        sentiment_change_score \
+            = 0  # will be amended once sentiment detection PR is added
+
+        textual_entailment_score \
+            = 0  # will be amended once textual entailment PR is added
+
+        unpicked_weight_cost_ratio_score \
+            = self._get_unpicked_weights_to_cost_ratio(stamp_page_index)
+
+        sweeping_index_score = self._get_distance_from_sweeping_index(
+            stamp_page_index)
+
+        return self._compute_weighted_score(
+            [
+                content_change_score,
+                sentiment_change_score,
+                textual_entailment_score,
+                unpicked_weight_cost_ratio_score,
+                sweeping_index_score
+            ]
+        )
+
+    def _update_sweeping_index(self):
+        # TODO: amend this to have variable
+        # change based on iteration
+        self.sweeping_index -= self.step_size
+
+    def _get_next_best_stamp_page_index(self):
+        ''' returns the index of the
+        next best stamp page to pick given knowledge
+        of the last picked stamp page
+        '''
+
+        if len(self.stamp_page_indices) == 0:
+            return -1
+
+        # sort based on interestingness metric
+        self.stamp_page_indices.sort(
+            key=self._interestingess_metric, reverse=True)
```

Comment:
How does sorting work here? Will the _interestingess_metric be calculated for every stamp_page comparison? Pre-calculating this score and storing it might be a better idea here.

Reply:
These are dependent on the last picked stamp page - hence change every iteration. 
They cannot be stored once and accessed again and again. Hope this helps!

Reply:
According to the article here: https://wiki.python.org/moin/TimeComplexity, the complexity for list sorting in python is O(nlogn). Which means there will be O(nlogn) comparisons? If I understand correctly, for each comparison, the interesting metric will be calculated for both the stamp pages.
IMO, we could minimize the number of times the interestingness metric is calculated by remembering this value for each stamp page in a given iteration.

Reply:
The "key" function specified in the sorting as a parameter is called only once before sorting and is remembered. 
Supporting answer: https://stackoverflow.com/a/52911342

Reply:
Awesome then!